rayon-core/src/test.rs

```diff
@@ -30,3 +33,53 @@ fn worker_thread_index() {
     let index = pool.install(|| pool.current_thread_index().unwrap());
     assert!(index < 22);
 }
+
+#[test]
+fn start_callback_called() {
+    let n_called = Arc::new(AtomicUsize::new(0));
+
+    let clone = n_called.clone();
+    let conf = Configuration::new()
+        .set_num_threads(16)
+        .set_start_handler(Arc::new(move |index| {
+            clone.fetch_add(1, Ordering::SeqCst);
+        }));
+    {
+        let pool = ThreadPool::new(conf).unwrap();
+    }
+
+    // We must have started at least one thread.
+    assert!(n_called.load(Ordering::SeqCst) > 0);
```

Comment:
but I think that personally I would be inclined to have them the startup cause messages to be sent, so that we can check that *all* threads start up. Though I'm not sure I'd want to guarantee that forever; but it *is* observable. Perhaps it's worth leaving a comment to the effect that threads may in the future start/stop at different times, but that whenever they start/stop, these handlers will be run.